config/environments/production.rb

```diff
@@ -16,7 +16,7 @@
 
   # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
   # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
-  # config.require_master_key = true
+  config.require_master_key = true
```

Comment:
yeah this will go back to a discuss @citizen428 and I had a bit ago about whether credentials are going to be a good pattern in general. it seems like there is a place for them, but it isn't clear to me what that place is off hand. You don't really want to give all of the production keys to everyone (and every system) that has access to the repo and the master key, which presumably means multiple creds files with multiple keys, and that kind of defeats the purpose of putting them in version control in the first place. And you don't want some things in ENV vars and some things in credentials, that seems to increase the cognitive overhead unnecessarily. We can re-hash this discussion here or elsewhere but we should probably decide if we recommend using the new rails credentials pattern for new apps in general, then we can make this kind of reinforcing template change

Reply:
> You don't really want to give all of the production keys to everyone (and every system) that has access to the repo and the master key

I'm not sure I follow. If you don't want someone to have access, don't give them the key. That's why you check in an encrypted file. 

Either way though, the longer I think about it, the more I'm leaning against using credentials. Most of the apps we build are running in rather simple environments (one Heroku instance), so managing through env vars is not too daunting a task, even if you need to do it for 2 or 3 environments. For bigger deployments this certainly becomes an issue, which is why e.g. Ansible has [Vault](https://docs.ansible.com/ansible/2.4/vault.html) which works in a similar way to credentials.